packages/pilot/src/containers/Reprocess/index.js

```diff
@@ -34,143 +34,172 @@ const validateOnRestart = ({ onRestart, stepStatus }, propName) => {
-                },
-                {
-                  id: 'result',
-                  title: t('pages.reprocess.result'),
-                },
-              ]}
-            />
-          </ModalSection>
-        </ModalContent>
-        {stepStatus
-          && stepStatus.result
-          && stepStatus.result !== 'pending'
-          ? (
-            <Result
-              onCopyIdClick={onCopyId}
-              onRestart={onRestart}
-              onViewTransactionClick={onViewTransaction}
-              status={stepStatus.result}
-              statusMessage={statusMessage}
-              t={t}
-            />
-          )
-          : (
-            <Form
-              amount={amount}
-              cardFirstDigits={first_digits} // eslint-disable-line camelcase
-              cardLastDigits={last_digits} // eslint-disable-line camelcase
-              error={
-                stepStatus.confirmation === 'error'
-                  ? statusMessage
-                  : ''
-              }
-              holderName={holder_name} // eslint-disable-line camelcase
-              loading={loading}
-              installments={installments}
-              onCancel={onCancel}
-              onConfirm={onConfirm}
-              t={t}
-            />
-          )
-        }
-      </Modal>
-    )
+const buildSteps = uncurryN(3, (stepStatus, translation) => juxt([
+  always(buildStep('identification', translation, stepStatus)),
+  when(
+    equals(true),
+    always(buildStep('confirmation', translation, stepStatus))
+  ),
+  always(buildStep('result', translation, stepStatus)),
+]))
+
+const getSteps = uncurryN(3, (stepStatus, translation) => pipe(
+  buildSteps(stepStatus, translation),
+  reject(equals(false))
+))
+
+const Reprocess = ({
+  allowReprocessWithoutAntifraud,
+  isOpen,
+  loading,
+  lockReason,
+  onBack,
+  onCancel,
+  onCopyId,
+  onForward,
+  onRestart,
+  onReprocess,
+  onViewTransaction,
+  statusMessage,
+  stepStatus = {},
+  t,
+  transaction: {
+    amount,
+    card: {
+      holder_name: holderName,
+    },
+    id: transactionId,
+  },
+}) => {
+  const [isWithoutAntifraud, setIsWithoutAntifraud] = useState(false)
+
+  const handleReprocess = withoutAntifraud => onReprocess({
+    amount,
+    transactionId,
+    withoutAntifraud,
+  })
+
+  const onReprocessTypeSelection = (withoutAntifraud) => {
+    setIsWithoutAntifraud(withoutAntifraud)
+
+    if (allowReprocessWithoutAntifraud) {
+      return onForward()
+    }
+
+    return handleReprocess(isWithoutAntifraud)
   }
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      loading={loading}
+      onRequestClose={onCancel}
+    >
+      <ModalTitle
+        closeIcon={<IconClose height={16} width={16} />}
+        onClose={onCancel}
+        title={t('pages.reprocess.title')}
+      />
+      <ModalContent>
+        <ModalSection>
+          <Steps
+            status={getSteps(stepStatus, t, allowReprocessWithoutAntifraud)}
```

Comment:
Esta declaraçãp poderia estar em uma variável fora deste return pois a declaração é repetida na linha abaixo.
```suggestion
           const steps = getSteps(stepStatus, t, allowReprocessWithoutAntifraud)}
```

Reply:
Feito!